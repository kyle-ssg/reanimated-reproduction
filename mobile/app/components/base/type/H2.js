/**
 * Created by kylejohnson on 11/05/2016.
 */
/**
 * Created by kylejohnson on 04/05/2016.
 */
const h2 = class extends React.Component {
    render() {
        return (
            <Text style={[Styles.h2, this.props.style]}>{this.props.children}</Text>
        );
    }
};

h2.propTypes = {
    style: React.PropTypes.any,
    children: OptionalElement,
};

module.exports = h2;
