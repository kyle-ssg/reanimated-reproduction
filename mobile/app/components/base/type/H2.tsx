import React from "react";
import { Text, TextStyle } from 'react-native';
import Styles from '../../../style/_style_screen';

interface Props {
  accessible?: boolean;
  accessibilityLabel?: string;
  style?: TextStyle;
  children?: React.ReactNode;
}

const h2: React.FC<Props> = ({
  accessible,
  accessibilityLabel,
  style,
  children,
}) => (
  <Text
    accessible={accessible}
    accessibilityLabel={accessibilityLabel}
    style={[Styles.h2, style]}
  >
    {children}
  </Text>
);

export default h2;
