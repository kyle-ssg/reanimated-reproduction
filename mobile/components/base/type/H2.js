/**
 * Created by kylejohnson on 11/05/2016.
 */
/**
 * Created by kylejohnson on 04/05/2016.
 */
module.exports = class extends React.Component {
    render () {
        return (
            <Text style={[Styles.h2, this.props.style]}>{this.props.children}</Text>
        )
    }
};
