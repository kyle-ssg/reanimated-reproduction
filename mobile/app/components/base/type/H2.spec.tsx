import * as React from "react";
import H2 from "./H2";
import renderer from "react-test-renderer";


describe("<H2 />", () => {
  it(`renders correctly`, () => {
    const tree = renderer.create(<H2>Example Title</H2>);
    expect(tree).toMatchSnapshot();
  });
});
