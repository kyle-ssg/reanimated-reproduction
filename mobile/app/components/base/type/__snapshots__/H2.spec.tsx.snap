// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<H2 /> renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "System",
        "fontSize": 23.2,
      },
      undefined,
    ]
  }
>
  Example Title
</Text>
`;
